src/OrchardCore.Modules/OrchardCore.ContentFields/Views/TextField-PredefinedList.Edit.cshtml

```diff
@@ -0,0 +1,18 @@
+@model OrchardCore.ContentFields.ViewModels.EditTextFieldViewModel
+@using OrchardCore.ContentFields.Settings
+
+@{
+    var settings = Model.PartFieldDefinition.GetSettings<TextFieldPredefinedListEditorSettings>();
+
+    // Morph the current shape to the correct editor shape based on its Editor property
+    var shape = Model as OrchardCore.DisplayManagement.IShape;
+    shape.Metadata.Alternates.Clear();
+    shape.Metadata.Type = "TextField_Editor__PredefinedList";
```

Comment:
Do we still need this trick since the changes we did for editors?

Reply:
There is a new method to compute the shape type based on the editor in the base class. Look at the all the fields `GetShapeType` so I don't think this kind of template is necessary anymore.

Reply:
With the ContentFieldDisplayDriver you can do GetEditorShapeType() but here I'm morphing a shape that is coming from the ContentPartFieldDefinitionDisplayDriver itself. I'm looking right now if I can add a GetEditorShapeType() method to compute that shape type name.